core/common/src/ByteStrings.kt

```diff
@@ -117,11 +117,24 @@ public fun Source.indexOf(byteString: ByteString, startIndex: Long = 0): Long {
     return -1
 }
 
+/**
+ * Returns the index of the first match for [byteString] in the buffer at or after [startIndex].
+ * Returns `-1` if the stream is exhausted before the requested bytes are found.
+ *
+ * For empty byte strings this function always return `0`.
+ *
+ * @param byteString the sequence of bytes to find within the buffer.
+ * @param startIndex the index into the buffer to start searching from.
+ *
+ * @throws IllegalArgumentException if [startIndex] is negative or if it is greater of equal to buffer's [Buffer.size].
```

Comment:
BTW, this behavior is also inconsistent with `CharSequence.indexOf`, which I'm considering maybe not as a paragon, but as something to be aligned with.